### PR TITLE
[DOC] Removing User Guide pages that will be source of truth on docs.nvidia…

### LINKED ISSUE #9249: [DOC] Deprecate user guide pages in Github now that docs.nvidia.com User Guide is available
The User Guide for Spark RAPIDS on docs.nvidia.com has launched to direct users for adopting Spark RAPIDS with basics on the plugin, environment setup, basic configuration, and tuning.  Documentation that is tied to advanced features and contributors will remain on Github.

Tasks
- [x] Make sure any commits from August 30th to present in github User Guide pages are reflected in docs.nvidia.com User Guide: https://github.com/NVIDIA/spark-rapids/commits/branch-23.10/docs
- [x] Remove all user guide pages from Github: Overview, all Getting Started pages, Configuration, Tuning, Qualification Tool, Profiling Tool, Additional Functionality (RAPIDS Accelerator for Apache Spark ML Library Integration, RAPIDS Shuffle Manager, Apache Iceberg Support, Delta Lake Support, RAPIDS Accelerator File Cache), Frequently Asked Questions, Examples, Contact Us
- [x] Add a link to the docs.nvidia.com User Guide (https://docs.nvidia.com/spark-rapids/) at the overview page for the github docs

### PR DESCRIPTION
**Pull request by mattahrens**
Closes #9249 

Removed User Guide pages that will now live at docs.nvidia.com: https://docs.nvidia.com/spark-rapids/user-guide/latest/index.html

### CODE REVIEW

**Comment by mattahrens** [comment]
build

File: docs/index.md [hunk 1]
```diff
@@ -16,37 +16,4 @@ a new framework to process data quickly and cost-efficiently with GPUs.
...
-
-## A Unified AI framework for ETL + ML/DL 
-A single pipeline, from ingest to data preparation to model training
-![spark3cluster](/docs/img/spark3cluster.png)
-
-## Technical Support 
-
-If you need any help or have questions on this product, please contact us : 
-spark-rapids-support@nvidia.com
+If you are a customer looking for information on how to adopt Spark RAPIDS for your Spark workloads, please go to our User Guide for more information: <a href="https://docs.nvidia.com/spark-rapids/user-guide/latest/index.html">link</a>.
```
Comment (jlowe) [inline docs/index.md:19]: Word wrap to approx. 100 columns.

"Spark RAPIDS" should be "the RAPIDS Accelerator for Apache Spark"

**Review by jlowe — COMMENTED** [review]
README needs to be updated, since it now refers to things that have been removed.

File: docs/index.md [hunk 2]
```diff
@@ -16,37 +16,5 @@ a new framework to process data quickly and cost-efficiently with GPUs.
...
-## A Unified AI framework for ETL + ML/DL 
-A single pipeline, from ingest to data preparation to model training
-![spark3cluster](/docs/img/spark3cluster.png)
-
-## Technical Support 
-
-If you need any help or have questions on this product, please contact us : 
-spark-rapids-support@nvidia.com
+If you are a customer looking for information on how to adopt RAPIDS Accelerator for Apache Spark
+for your Spark workloads, please go to our User Guide for more information: <a href="https://docs.nvidia.com/spark-rapids/user-guide/latest/index.html">link</a>.
```
Comment (jlowe) [inline docs/index.md:20]: Nit: Would be nice to have consistency in how we do links.
```suggestion
for your Spark workloads, please go to our User Guide for more information:
[link](https://docs.nvidia.com/spark-rapids/user-guide/latest/index.html).
```

**Review by jlowe — COMMENTED** [review]
[link checker workflow](https://github.com/NVIDIA/spark-rapids/actions/runs/6382677138/job/17321789830?pr=9362) is flagging a number of other broken links coming from other docs.  RapidsConf.scala will need to be updated among other docs.

File: docs/index.md (hunk 2 above)
Comment (mattahrens) [reply docs/index.md]: OK, updated link style for index.md page and fixed broken links

**Comment by mattahrens** [comment]
build

**Review by jlowe — APPROVED** [review]

**Comment by mattahrens** [comment]
build

**Comment by mattahrens** [comment]
build

**Pull request merged**